From e8ab4858628e86c36cc60257eaa17fd635ffd32d Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 18 Sep 2022 11:41:16 +0300
Subject: [PATCH] [SW-0] uart: add preboot_puts_hex64_raw() function

The new preboot_puts_hex64_raw() function is printing the provided
string and afterwards the unsigned long long value in hexadecimal
base format.
The EOL is not printed, as the function name suggests ("raw").

Change-Id: Ibe88c8b0108ea292457fef85c7748824be36e3a4
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/lib/common/uart.c b/lib/common/uart.c
index 130530a..81bc783 100644
--- a/lib/common/uart.c
+++ b/lib/common/uart.c
@@ -233,6 +233,15 @@
 	preboot_puth_raw((val >> 32), pad);
 	preboot_puth((val & 0xffffffff), true);
 }
+
+void preboot_puts_hex64_raw(const char *log, unsigned long long val, bool pad)
+{
+	preboot_puts_raw(log);
+	preboot_puts_raw("0x");
+	preboot_puth_raw((val >> 32), pad);
+	preboot_puth_raw((val & 0xffffffff), true);
+}
+
 void preboot_puts_char(const char *str, unsigned char val)
 {
 	preboot_puts_hex(str, val, false);
diff --git a/lib/common/uart.h b/lib/common/uart.h
index de4de38..2919257 100644
--- a/lib/common/uart.h
+++ b/lib/common/uart.h
@@ -20,6 +20,7 @@
 void preboot_puts_hex_raw(const char *log, unsigned int val, bool pad);
 void preboot_puts_hex(const char *log, unsigned int val, bool pad);
 void preboot_puts_hex64(const char *log, unsigned long long val, bool pad);
+void preboot_puts_hex64_raw(const char *log, unsigned long long val, bool pad);
 void preboot_puts_char(const char *str, unsigned char val);
 bool preboot_uart_tx_full(void);
 void preboot_uart_tx_flush(void);
